packages/rainbowkit/src/wallets/Wallet.ts

```diff
@@ -7,13 +7,12 @@ export type InstructionStepName =
   | 'connect'
   | 'refresh';
 
-type RainbowKitConnector<C extends Connector = Connector> = {
-  connector: C;
+type RainbowKitConnector = {
   mobile?: {
-    getUri?: () => Promise<string>;
+    getUri?: (uri: string) => string;
   };
   desktop?: {
-    getUri?: () => Promise<string>;
+    getUri?: (uri: string) => string;
     instructions?: {
       learnMoreUrl: string;
       steps: {
```

Comment:
Reasons for this change & approach we're taking from here:

1. You cannot create the connector first and then prepare to get the URI.
Instead, you would need to pass in a function (`CreateConnectorFn`) to wagmi's `createConfig`.

2. In the React tree, we can ensure that the URI is obtained from the connector. Once we've done that, we can then convert the URI to the format specified in the RainbowKit connector for the `mobile.getUri`, `desktop.getUri` and `qrCode.getUri`.